### ISSUE TITLE
Plugin for Digital Enlightenment / FX5 interfaces

### ISSUE THREAD

**Issue by ola-importer** [OpenLightingProject/ola#241]
_From [marius.z...@mlba-team.de](https://code.google.com/u/100596486661599741433/) on June 15, 2013 03:07:46_

Added plugin support for ubs dmx interfaces coming from digital enlightenment.

The necessary changes are contained in our github repository located at https://github.com/mlba-team/open-lighting , changesets ba94b3b43c6d846fb18c155530199e64c02eb5ac and a750505367f0d4f122b5affc93da3b09a4c44449. (See the diff attached as well)

The plugin depends on the libusbdmx driver from fx5, forked at https://github.com/mlba-team/usbdmx . Detection via autoconf was hopefully implemented correctly.

Please have a look at the changes and merge them when useful.

Thanks!

**Attachment:** [0001-Merged-support-for-digital-enlightenment-fx5-interfa.patch 0001-Fixed-missing-generation-of-Makefile.im-for-the-Enli.patch](http://code.google.com/p/open-lighting/issues/detail?id=240)

_Original issue: http://code.google.com/p/open-lighting/issues/detail?id=240_

**Comment by bitte-ein-bit** [comment]
I've noticed that the DMXControl Noodle U1 is listed as supported. 
According to https://github.com/DMXControl/usbdmx/commit/89bc53755711236ce5fd26977f08e223df31b980 the Noodle U1 is using the same driver/protocol, if I understand right. Or at least at that time was doing so. 
I fail to find the code where the USB identifiers are stored, so I can test. Any hints?

**Comment by peternewman** [comment]
Hi @bitte-ein-bit ,

Well spotted.

Assuming you mean on the OLA side, it's here:
https://github.com/OpenLightingProject/ola/blob/master/plugins/usbdmx/DMXCProjectsNodleU1Factory.cpp#L33-LL42

Have you got a Digital Enlightenment/FX5 interface then?

**Comment by bitte-ein-bit** [comment]
Yeah, I have a DE interface. I'll test and report back. Sometimes upper vs. lowercase search is to simple. ;-)

**Comment by peternewman** [comment]
So that's this line here:
https://github.com/OpenLightingProject/ola/blob/master/plugins/usbdmx/AsyncUsbSender.cpp#L67

Which looking at https://github.com/libusb/libusb/blob/master/libusb/libusb.h#L1105 I think will be transfer timed out.

Have you got another Linux machine you could try it on? Is your Pi overclocked? Can you try increasing the nice level of the process?

Is it just that one update that fails, or all subsequent ones too?

**Comment by peternewman** [comment]
You could also try increasing https://github.com/bitte-ein-bit/ola/blob/master/plugins/usbdmx/DMXCProjectsNodleU1.cpp#L52 and see if it helps stop the issue occurring.

**Comment by mkersting** [comment]
Is there any news to the DE Interface? 

I managed OLA to work on my Raspberry PI,
I managed to see the logs when in Plug and Unplug my Interface. 
I manages to run the python script which comes with the DE-Interface Driver And sucessfully gave out some DMX Signals. 

But I did not manage to add the Interface to OLA. 
ATM for me quite Frustrating, to look through Forum posts, some of them saying that they made it work (but no closer description). Others saying that the DE and FX5 Interface are not supported anymore.
But the Nodle is, and what is codewise the difference to the DE and FX5 Interface? 

I would be very happy to have either an old OLA Version where it works, or some News to the latest Information about this topic.

If I manage to make it work, i promise to write a "How to" Step by Step Guide :-D 

In the End of my personal Project I'd Like to receive ArtNet with the RaspberryPi and send it to the Output of the DE-Interface Which makes my Raspberry Pi an "ArtNet Node". So any other Suggestions to solve this are also very welcome :-)
